lib/sshkit/ssh/channel.ex

```diff
@@ -39,6 +39,31 @@ defmodule SSHKit.SSH.Channel do
     end
   end
 
+  @doc """
+  Activates a subsystem on a channel.
+
+  On success, returns `:ok`. On failure, returns `{:error, reason}`.
```

Comment:
Not entirely sure about `{:error, reason}` here - you're implementation always defaults to a `:failure` atom. Would the underlying Erlang implementation return more information?

Reply:
I always have a hard time reading the Erlang docs, so it might be worth double checking this. But this seems to be correct, the case statement is a bit odd because the Erlang function is odd in returning either `:failure` or `{:error, reason}` for failures. but the results become more consistent here:
This way we'd return `:success` or `{:error, :failure}`, `{:error, :closed}` or `{:error, :timeout}`, not very descriptive errors, but that's all that's available.

Reply:
I just took a look at the tests and their mocks and noticed that all the channel functions match the Erlang responses, a lot of the channel functions have the `:success | :failure | {:error, :timeout} | {:error, :closed}` response. I still think that's an odd response, but currently `Channel.subsystem` is the only exception so simply removing the case statement from the function would make it consistent with the rest.